loguru.hpp

```diff
@@ -102,6 +102,35 @@ Website: www.ilikebigbits.com
 #include <sal.h>	// Needed for _In_z_ etc annotations
 #endif
 
+/*
+	Will add syslog as a standard sink for log messages
+	Any logging message with a verbosity lower or equal to
+	the given verbosity will be included.
+
+	This works for Unix like systems (i.e. Linux/Mac)
+	There is no current implementation for Windows (as I don't know the
+	equivalent calls or have a way to test them). If you know please
+	add and send a pull request.
+
+	The code should still compile under windows but will only generate
```

Comment:
"The code should still compile under windows"

Can you give the motivation for this?

As the PR is written, the user can just call `add_syslog` and get a warning printed on windows. They will probably want to silence that warning, and so may end up wrapping their call to `add_syslog` with e.g. `#if LOGURU_SYSLOG` or `#ifdef __linux__`.

The alternative is to only declare `add_syslog` on systems that support it. That would force the user to wrap it in e.g. `#if LOGURU_SYSLOG`.

My instinct would be to go for the second alternative to force the user to consider syslog actively, but I'm not sure.

Reply:
I would argue against that.
But don't care that much about WIndow Users ;-) to make a big deal.

Why would you wrap the call to silence the warning?
Its one line in the log file. And I would assume that there is an equivalent in Windows that somebody is going to implement. At which point the log line would change to indicate that it works.

The main reason to have a windows version is to keep the interface symmetric.
If you are writing multi-platform code you don't want to have code that has a different API on different platforms. Hand hacking your code with `#if` is not a good plan is it means that at some point in the future you need to fix your code and remove the conditional code (tech-debt is real and leaving dirty code is a code smell) when loguru is updated (you would want the conditional code to be in loguru) so when it is fixed you don't change your code you just update the loguru library.

Reply:
I don't care much about Windows either BTW :P